mapflow/functional/service/data_catalog.py

```diff
@@ -28,13 +32,22 @@ class DataCatalogService(QObject):
     def __init__(self,
                  http: Http,
                  server: str,
-                 dlg: MainDialog):
+                 dlg: MainDialog,
+                 iface):
         super().__init__()
         self.dlg = dlg
         self.api = DataCatalogApi(http=http, server=server)
         self.view = DataCatalogView(dlg=dlg)
         self.mosaics = {}
-        
+        self.result_loader = layer_utils.ResultsLoader(iface=iface,
```

Comment:
Isn't it better to pass the main Mapflow's resultLoader here as an argument? So that we could avoid setting two temp dirs and two instances of the service

Reply:
Wouldn't it be circular import?
Plus, Mapflow's result_loader is still layer_utils.ResultsLoader. 
Or are you talking about something else?